arches_resource_sets/views/resource_set_view.py

```diff
@@ -0,0 +1,66 @@
+from json import JSONDecodeError
+import logging
+import uuid
+
+from django.http import Http404
+
+from arches.app.views.api import APIBase
+from arches_resource_sets.models import ResourceSet
+from arches.app.utils.response import JSONErrorResponse, JSONResponse
+from arches.app.utils.betterJSONSerializer import JSONDeserializer, JSONSerializer
+
+logger = logging.getLogger(__name__)
+
+class ResourceSetView(APIBase):
+    def get(self, request, set_id=None):
+        if set_id is None:
+            return JSONResponse({"resource_sets": ResourceSet.objects.all()})
+        set = ResourceSet.objects.filter(id=set_id).values().first()
+        return JSONResponse({"resource_set": set})
+
+    def put(self, request, set_id):
+        try:
+            set = ResourceSet.objects.get(id=set_id)
+            request_body = JSONDeserializer().deserialize(request.body)
+            description = request_body["description"] if "description" else ""
+            set.description = description
+            set.save()
+            return JSONResponse({"resource_set": set})
+        except ResourceSet.DoesNotExist:
+            return JSONErrorResponse("Could not update resource set", "Resource set id '{}' not found".format(set_id), status=404)
+        except JSONDecodeError:
+            return JSONErrorResponse("Could not update resource set", "Invalid request body: '{}'".format(request.body), status=400) 
+        except Exception as e:
+            logger.error(e)
+            return JSONErrorResponse("Unspecified error", "Contact an Administrator", status=500)
+
+
+
+    def post(self, request):
+        set_id = uuid.uuid4()
+        set = ResourceSet.objects.create(id=set_id, owner=request.user)
+        try:
+            request_body = JSONDeserializer().deserialize(request.body)
+            description = request_body["description"] if "description" in request_body else ""
+            set.description = description
+            set.save()
+            return JSONResponse({"resource_set": set})
+        except JSONDecodeError:
+            return JSONErrorResponse("Request was malformed", "Ensure the request body is valid json", status=400)
+        except Exception as e:
+            logger.error(e)
+            return JSONErrorResponse("Unspecified error", "Contact an Administrator", status=500)
+
+
+
+    def delete(self, request, set_id):
+        try:
+            set_obj = ResourceSet.objects.get(id=set_id)
+            set_obj.delete()
+            return JSONResponse({"id":set_id, "resource_set": set_id})
+        except ResourceSet.DoesNotExist:
```

Comment:
```suggestion
            set_obj.id = set_id
            return JSONResponse({"resource_set": set_obj})
```

Reply:
I'm not sure what you're suggesting here, but I agree that the return needs to be tweaked.  I modified it to return the object that was deleted in the same way we do in all the other methods.

Reply:
I noticed that the `set_obj` object didn't include the id, so I just added that back on with `set_obj.id = set_id`.

Reply:
oh neat.  Didn't realize that .delete() sets the uuid of the object to null.